### ISSUE TITLE
Dependency get's removed from lock file

### ISSUE THREAD

**Issue by kasium** [python-poetry/poetry#5351]
- [X] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [X] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Linux
- **Poetry version**: 1.1.13

## Issue

I have the following in my dependency list
```toml
[tool.poetry.dev-dependencies]
diff-cover = {version = "==6.4.4", extras = ["toml"]}
```

diff-cover depends on setuptools if you run on python < 3.8.
However, a install doesn't install setuptools and it's not listed in the lock file as a depdency of diff-cover
```toml
[[package]]
name = "diff-cover"
version = "6.4.4"
description = "Run coverage and linting reports on diffs"
category = "dev"
optional = false
python-versions = ">=3.6.2,<4.0.0"

[package.dependencies]
chardet = ">=3.0.0"
Jinja2 = ">=2.7.1"
pluggy = ">=0.13.1,<2"
Pygments = ">=2.9.0,<3.0.0"
tomli = {version = ">=1.2.1,<2.0.0", optional = true, markers = "extra == \"toml\""}
```

However, if using poetry 1.2.0b1, the following line will appear in the lock file
`setuptools = {version = ">=17.0.0", markers = "python_version < \"3.8\""}`

I report this bug even if (maybe) fixed, since 1.1 doesn't include it

**Comment by finswimmer** [comment]
Hello @kasium,

please see https://github.com/python-poetry/poetry/issues/1584 for the whole story. Especially https://github.com/python-poetry/poetry/issues/1584#issuecomment-567628022 and https://github.com/python-poetry/poetry/issues/1584#issuecomment-589644457.

TL;DR: Poetry haven't locked `setuptools` explicit before version 1.2 and will do it now.

fin swimmer

**Comment by kasium** [comment]
thanks a lot!
